common/dbconfig.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021-present Pinterest. All Rights Reserved.
+//
+// @author Premkumar (prem@pinterest.com)
+//
+
+#include "common/dbconfig.h"
+#include "common/segment_utils.h"
+#include <utility>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+
+DECLARE_int32(replicator_replication_mode);
+
+namespace common {
+
+DBConfigManager* DBConfigManager::get() {
+  static DBConfigManager instance;
+  return &instance;
+}
+
+DBConfigManager::DBConfigManager() : dbConfig_(new DBConfig()), fDataLoaded_(false) {
+}
+
+bool DBConfigManager::loadJsonStr(const std::string& content) {
+  Json::Value root;
+  Json::Reader reader;
+  if (!reader.parse(content, root) || !root.isObject()) {
+    LOG(ERROR) << "Could not load config";
+    return false;
+  }
+
+  return loadJsonObject(root);
+}
+
+bool DBConfigManager::loadJsonObject(const Json::Value& root) {
+  try {
+    auto newConfig = parseConfig(root);
+    if (newConfig == nullptr) {
+      return false;
```

Comment:
nit: when there is no config available (e.g. the config file is not available yet), this mean we'll return `false`. Can you clarify this in the API comments, otherwise it may be a little confusing to the reader, who wonder if there is difference between:
- config is not available, or empty
- config is failed to load/parse

Reply:
sounds good, Jia